HealthcareSystem/GraphicalEditorServer/Controllers/EquipmentTypeController.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Backend.Model.Exceptions;
+using Backend.Model.Manager;
+using Backend.Service.RoomAndEquipment;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace GraphicalEditorServer.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class EquipmentTypeController: ControllerBase
+    {
+        private readonly IEquipmentTypeService _equipmentTypeService;
+
+        public EquipmentTypeController(IEquipmentTypeService equipmentTypeService)
+        {
+            _equipmentTypeService = equipmentTypeService;
+        }
+
+        [HttpPost]
+        public ActionResult AddEquipmentType([FromBody] EquipmentType equipmentType)
+        {
+                EquipmentType addedEquipmentType=_equipmentTypeService.AddEquipmentType(equipmentType);            
+                 return Ok(addedEquipmentType.Id);
+        }
+
+        [HttpGet]
+        public ActionResult GetAllEquipmentTypes()
+        {
+            List<EquipmentType> equipmentTypes = new List<EquipmentType>();
+            try
+            {
+                equipmentTypes = _equipmentTypeService.GetEquipmentTypes();
+                return Ok(equipmentTypes);
+            }
+            catch (NotFoundException exception)
+            {
+                return NotFound(exception.Message);
+            }
```

Comment:
What would throw a NotFound exception here?

Reply:
I am not sure if this method will ever throw exception, but if it does it should mean that there were exception while getting allEquipmentTypes, l guess nothing more precisely then that. I made that method based on other get methods that we previously done. Do you think l should change something there?

Reply:
Yeah, so, this is something I've been discussing with the Web team quite a bit recently, since they had quite a few issues with their error handling (check out PR #91 and PR #99 if you're interested). Basically, in this particular case, NotFoundException is one of our internal exceptions, correct? You only need to catch that if you explicitly throw it somewhere as a message for your higher layers and want to catch it in the Controller for example to deal with that message, but it's not going to come out of nowhere. So since I don't think you throw a NotFoundException anywhere in your Service and Repository, this is never going to catch anything. This call specifically *could* throw exceptions, but they'll actually be exceptions from various Entity Framework failures since you don't have any try catch blocks in the Repository to catch those and either handle them somehow, or replace them with your own exceptions with precise meanings.
But yeah, if what you're going for here is to return 404 when no equipment types are found, a more accurate way to do that here would just be to check if the list you are getting is empty, and return 404 if it is.
I'm not really insisting you change anything here though, error handling is kind of a large topic that I don't think you've really done anything with in GraphicalEditorServer so far. If you want to tackle that you should probably discuss with your team (and possibly the Web team as well) exactly what your goals are with it and how you want to proceed.